core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -334,10 +334,18 @@ angular.module('oppia').component('contributionsAndReview', {
       ctrl.switchToTab = function(tabType, suggestionType) {
         ctrl.activeSuggestionType = suggestionType;
         ctrl.activeTabType = tabType;
+        ctrl.activeDropdownTabChoice =
+          ctrl.activeTabType + ' ' + ctrl.activeSuggestionType;
```

Comment:
Instead of doing string concat all over the place, make a function that handles this for you and use it everywhere. Something like getTabOptionNameFromXXX(tabType, suggestionType) (not sure what "XXX" should be).

That way, you encapsulate the logic in a single place. Put that new function next to whatever function you're using to extract the individual parts from the composite string, too (so, similarly, do not have .split() all over the place).

Reply:
@seanlip I've added functions to return a concatenated string (takes two arguments - tabType, suggestionType) and split the string `activeDropdowntabChoice` (as that is the only string I'm ever splitting) and return its constituent parts, and used these functions where necessary.

(I failed to get this when you first pointed it out, but now that I've made the changes, I understand why)

PTAL!